### PR TITLE
Throw Instead of Assert in PassValidation

### PR DESCRIPTION
**Pull request by mroethlin**
## Technical Description

Exceptions give nicer error messages in Jupyter Notebooks and, as opposed to assertions, also work in dawns Release mode. The error handling in dawn needs an overhaul either way. I don't know if bubbling up an exception is the proper way to go, however, it does seem more reasonable in instances for syntax errors and the like. Hence, all assertions in the validation pass were replaced by throwing semantic exceptions. This could serve as a stepping stone to further improve dawn's error handling.

### CODE REVIEW

**Comment by mroethlin** [comment]
launch jenkins

**Comment by mroethlin** [comment]
launch jenkins

File: dawn/src/dawn/Optimizer/PassValidation.cpp [hunk 1]
```diff
@@ -95,24 +84,25 @@ bool PassValidation::run(const std::shared_ptr<dawn::SIR>& sir) {
   if(sir->GridType == ast::GridType::Unstructured) {
     auto [checkResultDimensions, errorLocationDimensions] =
         UnstructuredDimensionChecker::checkDimensionsConsistency(*sir);
-    DAWN_ASSERT_MSG(checkResultDimensions, ("Dimensions in SIR are not consistent at line " +
-                                            std::to_string(errorLocationDimensions.Line))
-                                               .c_str());
+    if(!checkResultDimensions)
+      throw SemanticError("Dimensions in SIR are not consistent at line " +
+                          std::to_string(errorLocationDimensions.Line));
+    ;
```
Comment (Stagno) [inline dawn/src/dawn/Optimizer/PassValidation.cpp:90]:
```suggestion
    
```

**Review by Stagno — APPROVED** [review]
Good to go

**Comment by mroethlin** [comment]
launch jenkins

**Pull request merged**